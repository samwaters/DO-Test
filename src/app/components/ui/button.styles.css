.btn {
    border-radius: 5px;
    border: none;
    cursor: pointer;
}

.btn-small {
    font-size: 12px;
    padding: 5px 10px;
}

.btn-medium {
    font-size: 16px;
    padding: 10px 20px;
}

.btn-large {
    font-size: 20px;
    padding: 20px 30px;
}

.btn-primary {
    background-color: var(--primary);
    color: var(--text)
}

.btn-secondary {
    background-color: var(--secondary);
    color: var(--text);
}