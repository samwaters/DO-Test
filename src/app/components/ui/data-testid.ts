export interface PropsWithDataTestId {
    dataTestId?: string
}
