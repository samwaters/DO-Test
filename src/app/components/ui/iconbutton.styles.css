.icon-button {
    align-items: center;
    background-color: var(--primary);
    border: 1px solid var(--secondary);
    border-radius: 5px;
    display: flex;
    justify-content: center;
}

.icon-button.small {
    font-size: 12px;
    height: 20px;
    width: 20px;
}
.icon-button.medium {
    font-size: 20px;
    height: 36px;
    width: 36px;
}
.icon-button.large {
    font-size: 26px;
    height: 48px;
    width: 48px;
}