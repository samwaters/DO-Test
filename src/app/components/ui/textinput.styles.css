.text-input {
    border: 1px solid var(--primary);
    border-radius: 5px;
    outline: none;
}

.text-input-fullwidth {
    width: 100%;
}

.text-input-small {
    font-size: 12px;
    padding: 5px 10px;
}

.text-input-medium {
    font-size: 16px;
    padding: 10px 15px;
}

.text-input-large {
    font-size: 20px;
    padding: 20px 30px;
}
