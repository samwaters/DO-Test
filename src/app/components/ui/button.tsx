import { MouseEvent, PropsWithChildren } from "react"
import "./button.styles.css"
import { PropsWithDataTestId } from "@/app/components/ui/data-testid"

interface Props extends PropsWithChildren, PropsWithDataTestId {
    onClick: (e: MouseEvent<HTMLButtonElement>) => void
    size: "small" | "medium" | "large"
    variant: "primary" | "secondary"
}
export const Button = ({ children, dataTestId, onClick, size, variant }: Props) => {
    return (
        <button className={`btn btn-${size} btn-${variant}`} data-testid={dataTestId} onClick={onClick}>
            {children}
        </button>
    )
}
