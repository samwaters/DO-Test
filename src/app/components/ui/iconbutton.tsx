import { ReactElement } from "react"
import "./iconbutton.styles.css"
import { Tooltip } from "react-tooltip"
import { PropsWithDataTestId } from "@/app/components/ui/data-testid"

interface Props extends PropsWithDataTestId {
    icon: ReactElement
    id: string
    onClick?: () => void
    size?: "small" | "medium" | "large"
    tooltip?: string
}

export const IconButton = ({ dataTestId, icon, id, onClick, size, tooltip }: Props) => {
    return (
        <>
            <button
                className={`icon-button ${size}`}
                data-testid={dataTestId}
                data-tooltip-id={id}
                data-tooltip-content={tooltip}
                onClick={onClick}
            >
                {icon}
            </button>
            {tooltip && <Tooltip id={id} />}
        </>
    )
}
