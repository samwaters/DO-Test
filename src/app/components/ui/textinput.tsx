import "./textinput.styles.css"
import { ChangeEvent, useEffect, useRef } from "react"
import { PropsWithDataTestId } from "@/app/components/ui/data-testid"

interface Props extends PropsWithDataTestId {
    autoFocus?: boolean
    fullWidth?: boolean
    onChange: (e: ChangeEvent<HTMLInputElement>) => void
    size: "small" | "medium" | "large"
    value: string | undefined
}

export const TextInput = ({ autoFocus, dataTestId, fullWidth, onChange, size, value }: Props) => {
    const ref = useRef<HTMLInputElement>(null)
    useEffect(() => {
        if (autoFocus && ref.current) {
            ref.current.focus()
        }
    }, [autoFocus])

    return (
        <input
            className={`text-input text-input-${size} ${fullWidth ? "text-input-fullwidth" : ""}`}
            data-testid={dataTestId}
            onChange={onChange}
            ref={ref}
            type="text"
            value={value}
        />
    )
}
