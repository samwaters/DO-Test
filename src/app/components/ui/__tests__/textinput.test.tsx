import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import { TextInput } from "@/app/components/ui/textinput"

describe("TextInput", () => {
    it("Renders different text inputs", () => {
        const change = jest.fn()
        render(
            <div data-testid={"textinput"}>
                <TextInput onChange={change} size="small" value="" />
                <TextInput onChange={change} size="medium" value="med" />
                <TextInput onChange={change} size="large" value="" />
            </div>,
        )
        expect(screen.getByTestId("textinput")).toMatchSnapshot()
    })
    it("Calls the change handler correctly", async () => {
        userEvent.setup()
        const change = jest.fn()
        render(<TextInput dataTestId="input" onChange={change} size="small" value="" />)
        expect(change).not.toHaveBeenCalled()
        await userEvent.type(screen.getByTestId("input"), "f")
        expect(change).toHaveBeenCalledWith(expect.objectContaining({ type: "change" }))
    })
})
