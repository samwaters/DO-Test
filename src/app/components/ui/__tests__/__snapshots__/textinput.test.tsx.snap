// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Renders different text inputs 1`] = `
<div
  data-testid="textinput"
>
  <input
    class="text-input text-input-small "
    type="text"
    value=""
  />
  <input
    class="text-input text-input-medium "
    type="text"
    value="med"
  />
  <input
    class="text-input text-input-large "
    type="text"
    value=""
  />
</div>
`;
